src/Core/ComponentValidator.vala

```diff
@@ -57,13 +57,12 @@ public class AppCenterCore.ComponentValidator : Object {
 
     private void parse_and_populate (string contents) {
         foreach (string line in contents.split ("\n")) {
-            if (line.has_prefix ("#")) {
-                continue;
-            }
-            
-            string token = line.strip ();
-            if (token != "") {
-                blacklist.add (token);
+            string[] tokens = line.split ("#");
+            if (tokens.length > 0) {
+                string id = tokens[0].strip ();
+                if (id != "") {
```

Comment:
This assumes the id does not contain a # character.   Not sure if that matters.

Reply:
My point was that something like `my#tageditor.desktop # comment` might cause unexpected results but I am not sure whether app names which include a `#` are likely (or possible).  The id will be added to the blacklist as `my`.

Reply:
According to the docs, component IDs may contain a `#`

>  The `<id>` tag is a unique identifier for this component. It must contain only ASCII characters, dots, hyphens and numbers. Spaces are not allowed.

https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#sect-Metadata-GenericComponent